pandas/core/internals.py

```diff
@@ -492,6 +492,14 @@ def _astype(self, dtype, copy=False, errors='raise', values=None,
                 return self.copy()
             return self
 
+        # numpy arrays don't handle generic timestamp dtypes well. Since
+        # we only use the nanosecond frequency, interpret generic timestamp
+        # dtypes as nanosecond frequency.
+        if dtype.name == "datetime64":
```

Comment:
this is way too hacky to put this here, this *should* be trapped in ``maybe_cast_to_datetime``, or if not, then let's create a generic way (via function there to do this).